dhcpv6/option_ntp_server.go

```diff
@@ -0,0 +1,58 @@
+package dhcpv6
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/u-root/uio/uio"
+)
+
+// NTPSuboptionSrvAddr is the value of NTP_SUBOPTION_SRV_ADDR according to RFC 5908.
+const NTPSuboptionSrvAddr byte = 1
+
+// OptNTPServer returns an NTP Server option as defined by RFC 5908.
+func OptNTPServer(ntpServers ...net.IP) Option {
+	return &optNTPServer{NTPServers: ntpServers}
+}
+
+type optNTPServer struct {
+	NTPServers []net.IP
+}
+
+// Code returns the option code
+func (op *optNTPServer) Code() OptionCode {
+	return OptionNTPServer
+}
+
+// ToBytes returns the option serialized to bytes.
+func (op *optNTPServer) ToBytes() []byte {
+	buf := uio.NewBigEndianBuffer(nil)
+	for _, ns := range op.NTPServers {
+		buf.WriteBytes([]byte{NTPSuboptionSrvAddr, net.IPv6len})
+		buf.WriteBytes(ns.To16())
+	}
+	return buf.Data()
+}
+
+func (op *optNTPServer) String() string {
+	return fmt.Sprintf("NTP: %v", op.NTPServers)
+}
+
+// parseOptNTPServer builds an optNTPServer structure
+// from a sequence of bytes. The input data does not include option code and length
+// bytes.
+func parseOptNTPServer(data []byte) (*optNTPServer, error) {
+	var opt optNTPServer
+	buf := uio.NewBigEndianBuffer(data)
+	for buf.Has(2 + net.IPv6len) {
+		suboptionPrefix := buf.CopyN(2)
+		if suboptionPrefix[0] != NTPSuboptionSrvAddr {
```

Comment:
I don't think this is good. It will fail to parse RFC compliant options.

Reply:
oh, damn, the two fields are 16 bits each, not 8. Is that what you mean?

Reply:
Not really, what I mean is that the sub option code can be != NTPSuboptionSrvAddr